business/apps.go

```diff
@@ -63,14 +125,63 @@ func (in *AppService) GetAppList(namespace string) (models.AppList, error) {
 			Name:         keyApp,
 			IstioSidecar: true,
 		}
-		labels := make(map[string][]string)
+		applabels := make(map[string][]string)
+		svcReferences := make([]*models.IstioValidationKey, 0)
 		for _, srv := range valueApp.Services {
-			joinMap(labels, srv.Labels)
+			joinMap(applabels, srv.Labels)
+			if linkIstioResources {
+				svcVirtualServices := kubernetes.FilterVirtualServices(*linkedResources[kubernetes.VirtualServices], srv.Namespace, srv.Name)
+				svcDestinationRules := kubernetes.FilterDestinationRules(*linkedResources[kubernetes.DestinationRules], srv.Namespace, srv.Name)
+				allFiltered := append(svcVirtualServices, svcDestinationRules...)
+				for _, a := range allFiltered {
+					ref := models.IstioValidationKey{
```

Comment:
There's a method for creating `IstioValidationKey`s. And it is good to use it to hide implementation details and consistency, I guess.

```go
models.BuildKey(a.GetObjectMeta().Name, a.GetObjectMeta().Namespace, a.GetTypeMeta().Kind)
```

Reply:
Well, I may argue :).
We don't have constructors for all models, as it's not the standard way in golang, but as this one exists, I'm +1 to update it.